packages/webapp/src/components/Form/MultiStepForm/WithStepperProgressBar.tsx

```diff
@@ -57,36 +64,28 @@ export const WithStepperProgressBar = ({
   onCancel,
   onGoForward,
   handleSubmit,
-  formState: { isValid },
+  formState: { isValid, isDirty },
+  setFormResultData,
 }: WithStepperProgressBarProps) => {
-  const { t } = useTranslation();
+  const [transition, setTransition] = useState<{ unblock?: () => void; retry?: () => void }>({
+    unblock: undefined,
+    retry: undefined,
+  });
 
   const isSummaryPage = hasSummaryWithinForm && activeStepIndex === steps.length - 1;
 
-  const isSummaryPageRef = useRef(isSummaryPage);
-
-  useEffect(() => {
-    isSummaryPageRef.current = isSummaryPage;
-  }, [isSummaryPage]);
-
   // Block the page transition
   // https://github.com/remix-run/history/blob/dev/docs/blocking-transitions.md
   useEffect(() => {
-    if (isSummaryPage) {
+    if (isSummaryPage || !isDirty) {
       return;
     }
     const unblock = history.block((tx) => {
-      if (window.confirm(`TODO: ${t('CANCEL_FLOW_MODAL.BODY')}`)) {
-        // Unblock the navigation.
-        unblock();
-
-        // Retry the transition.
-        tx.retry();
-      }
+      setTransition({ unblock, retry: tx.retry });
```

Comment:
I read the linked page but I don't know how you managed to get from that code to this component state solution -- totally brilliant 😍